### PR TITLE
Remove deprecated methods from QuantumCircuit

### PR DESCRIPTION
**Pull request by jakelishman**
### Summary

This removes the old `c?u[1-3]` methods from `QuantumCircuit`, along with the `combine` and `extend` methods.  The numbered-argument `UGate` alternatives had been deprecated since Terra 0.16, and the simpler forms of `compose` since 0.17.

The `U1Gate` class, and so on, are left in so the transpiler and equivalence rules can still support legacy backends with these as basis gates, but their privileged place as direct methods on `QuantumCircuit` is removed.

### Details and comments

I also cut out the test warning suppressions that were related to these methods.

### CODE REVIEW

**Review by Cryoris — COMMENTED** [review]
LGTM, always nice to clean up the code 😉 There's a lint error due to an unused import of `warnings` in the `test_extensions_standard` but otherwise this looks good!

**Comment by jakelishman** [comment]
Updated, thanks!

File: releasenotes/notes/remove-deprecated-circuit-methods-3e4eb27c4709ba12.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+upgrade:
+  - |
+    The :class:`.QuantumCircuit` methods ``u1``, ``u2``, ``u3``, and their
+    controlled variants ``cu1``, ``cu2`` and ``cu3`` have been removed following
```
[the remark this answers is not in the document]
Comment (jakelishman) [reply releasenotes/notes/remove-deprecated-circuit-methods-3e4eb27c4709ba12.yaml]: Thanks - I just made a mistake here where I'd assumed it was symmetrical.  In reality, there was no `cu2` gate, and there _was_ `mcu1`.  Fixed in c897093e8.

**Review by Cryoris — APPROVED** [review]

**Comment by jakelishman** [comment]
@Mergifyio requeue

**Pull request merged**